### ISSUE TITLE
Not Spawning on Custom/Mod Blocks

### ISSUE THREAD

**Issue by ecoay** [OreCruncher/Debris#4]
I want to use this for my wasteland survival modpack however the debris is not spawning on cracked sand. I did find some on some stone in a crevice.

**Comment by OreCruncher** [comment]
Conditions for spawning on a block:
- Block must not be a debris block
- It cannot be a leaf block
- The top side must be solid as reported by isSideSolid(EnumFacing.UP)

Which cracked sand is you have?  BoP?

**Comment by ecoay** [comment]
Yeah BoP unless i am setting density incorrectly. I want like 1 piece of debris per chunk pretty much. Debris will play a big part in the wasteland.

**Comment by ecoay** [comment]
I cant wait for new configurable models IE add some trash bags etc and custom json of each. My modpack centers around returning to earth thousands of years after it has been uninhabitable.

**Comment by OreCruncher** [comment]
The "density" is the number of attempts per chunk that Debris tries to place a pile of some sort underground (< Y 64).  An attempt can fail if the block selected is already occupied, or if the block beneath isn't side solid on the top.  Increasing the density increases the amount of debris piles placed while decreasing reduces the number.  So if you want 1 debris per chunk you need to tune it downwards some.

**Comment by ecoay** [comment]
So debris only places underground?

**Comment by OreCruncher** [comment]
It uses the average ground level information minus 2 that is reported for the dimension.  Typically it is Y = 64 -2 or 62.  For the normal Overworld this works out to be underground.

**Comment by ecoay** [comment]
Maybe possibility config option to make it spawn ground level? The idea in this instance is to allow another method of obtaining flint etc for basic tools since no ore in wasteland world.

**Comment by OreCruncher** [comment]
Maybe a Y level override that if set to some value would be used instead of ground level.  I think I would have to add an entry for allowing customization of invalid/valid spawn blocks as well.
